packages/api/controllers/hackathon_team_controller.go

```diff
@@ -0,0 +1,169 @@
+package controllers
+
+import (
+	"context"
+	"hub-backend/configs"
+	"hub-backend/models"
+	"hub-backend/responses"
+	"net/http"
+	"reflect"
+	"time"
+
+	"github.com/gofiber/fiber/v2"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var hackathonTeamCollection *mongo.Collection = configs.GetCollection(configs.DB, "hackathonTeam")
+
+func CreateHackathonTeam(c *fiber.Ctx) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+
+	bearer_token := c.Get("BEARER-TOKEN")
+
+	var team models.Team
+	defer cancel()
+
+	if bearer_token != configs.ReturnAuthToken() {
+		return c.Status(http.StatusUnauthorized).JSON(responses.MemberResponse{Status: http.StatusUnauthorized, Message: "error", Data: &fiber.Map{"data": "Unauthorized"}})
+	}
+
+	if err := c.BodyParser(&team); err != nil {
+		return c.Status(http.StatusBadRequest).JSON(responses.MemberResponse{Status: http.StatusBadRequest, Message: "error", Data: &fiber.Map{"data": err.Error()}})
+	}
+
+	if validationErr := validate.Struct(&team); validationErr != nil {
+		return c.Status(http.StatusBadRequest).JSON(responses.MemberResponse{Status: http.StatusBadRequest, Message: "error", Data: &fiber.Map{"data": validationErr.Error()}})
+	}
+
+	result, err := hackathonTeamCollection.InsertOne(ctx, team)
+	if err != nil {
+		return c.Status(http.StatusInternalServerError).JSON(responses.MemberResponse{Status: http.StatusInternalServerError, Message: "error", Data: &fiber.Map{"data": err.Error()}})
+	}
+
+	return c.Status(http.StatusCreated).JSON(responses.MemberResponse{Status: http.StatusCreated, Message: "success", Data: &fiber.Map{"data": result}})
+}
+
+func GetHackathonTeams(c *fiber.Ctx) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+
```

Comment:
Missing auth

Reply:
Done, also added auth to the other GET requests